### PR TITLE
Modernize the examples

### LINKED ISSUE #62: Modernize the examples
Similar to what was done for the tests.

Should we move the examples into the test folder? Or is there a reason to keep them separate?

### PR DESCRIPTION
**Pull request by jacobwilliams**
Fixes #62

### CODE REVIEW

**Comment by jacobwilliams** [comment]
Ready to merge. Really just cosmetic changes so they match the same style as the other tests.

**Review by awvwgk — APPROVED** [review]
Looks good to me.

File: examples/example_primes.f90 [hunk 1]
```diff
@@ -26,71 +29,68 @@ subroutine find_fit(data_x, data_y, expr, pars)
...
+use iso_fortran_env, only: nwrite => output_unit
+
 implicit none
 
-real(dp) :: pars(3)
-real(dp), parameter :: y(*) = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, &
-    37, 41, 43, 47, 53, 59, 61, 67, 71]
+real(wp) :: pars(3)
+real(wp), parameter :: y(*) = real([2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, &
+                                    37, 41, 43, 47, 53, 59, 61, 67, 71], wp)
```
Comment (certik) [inline examples/example_primes.f90:76]: Why do you need the `real([...], wp)` part? It seems it compiles without it also?

Comment (awvwgk) [reply examples/example_primes.f90:76]: It is more a matter of style here, but I think the explicit conversion is preferable over the implicit one. To make the conversion explicit with using `real([...], wp)` you can add the type directly to the array constructor with `[real(wp):: ...]`.

File: examples/example_primes.f90 [hunk 2]
```diff
@@ -26,71 +29,68 @@ subroutine find_fit(data_x, data_y, expr, pars)
...
-fvec(1) = iflag
-fvec = data_y - expr(data_x, x)
-end subroutine
+    subroutine fcn(m, n, x, fvec, iflag)
+    integer, intent(in) :: m
+    integer, intent(in) :: n
+    integer, intent(inout) :: iflag
+    real(wp), intent(in) :: x(n)
+    real(wp), intent(out) :: fvec(m)
+    fvec(1) = iflag ! Suppress compiler warning:
```
Comment (certik) [inline examples/example_primes.f90:56]:
```suggestion
    fvec(1) = iflag ! Suppress compiler warning
```

File: examples/example_primes.f90 (hunk 1 above)
Comment (certik) [reply examples/example_primes.f90:76]: I see. Generally Fortran converts integers to the correct real/kind correctly, so it's safe to use the implicit conversion I think.

Comment (jacobwilliams) [reply examples/example_primes.f90:76]: Yeah, in this case, it's not a big deal. But I rarely or ever use implicit conversions. I think they can get people in trouble and can be confusing to beginners. If you want a real, you should declare a real (`2.0_wp`, etc).

**Pull request merged**